lib/daemon.js

```diff
@@ -62,6 +64,16 @@ class Daemon {
    */
   async start () {
     for await (const event of this._externalSecretEvents) {
+      // Check if the externalSecret should be managed by this instance.
+      if (event.object.spec) {
+        const externalSecretMetadata = event.object.metadata
+        const externalSecretController = event.object.spec.controllerID
+        if (externalSecretController && this._instanceId !== externalSecretController) {
```

Comment:
So in this case if an ExternalSecret does not specify controllerId an instance which has an instanceId would still look at it. 
Maybe we should modify it like this:

```suggestion
        if ((this._instanceId || externalSecretController) && this._instanceId !== externalSecretController) {
```

So if an KES instance has an instanceId the instance would only deal with those with matching controllerId, so if you want a KES instance taking care of all those without controller id you would leave instance id empty.

I suppose that boils down to just comparing in the end 😄 
```suggestion
        if (this._instanceId !== externalSecretController) {
```